text/0020-bech32-address-format/0020-bech32-address-format.md

```diff
@@ -0,0 +1,87 @@
++ Start date: 2020-07-28
++ RFC PR: [iotaledger/protocol-rfcs#0020](https://github.com/iotaledger/protocol-rfcs/pull/20)
+
+# Summary
+
+This document proposes an extendable address format for the IOTA protocol supporting various signature schemes and address types. It relies on the [Bech32](https://github.com/bitcoin/bips/blob/master/bip-0173.mediawiki) format to provide a compact, human-readable encoding with strong error correction guarantees.
+
+# Motivation
+
+IOTA uses the [Winternitz one-time signature scheme](https://docs.iota.org/docs/getting-started/1.0/cryptography/signatures) (W-OTS) to generate digital signatures, in which addresses correspond to a [Kerl](https://github.com/iotaledger/kerl) hash. With the introduction of Ed25519 signatures as part of [Chrysalis](https://roadmap.iota.org/chrysalis), it is necessary to define a new universal and extendable address format capable of encoding these two types of addresses.
+
+The current IOTA protocol relies on Base27 addresses with a truncated Kerl checksum. However, both the character set and the checksum algorithm have limitations: 
+- Base27 is designed for ternary and is ill-suited for binary data.
+- The Kerl hash function also requires ternary input. Further, it is slow and provides no error-detection guarantees.
+- It does not support the addition of version or type information to distinguish between different kinds of addresses with the same length.
+
+All of these points are addressed in the Bech32 format introduced in [BIP-0173](https://github.com/bitcoin/bips/blob/master/bip-0173.mediawiki): In addition to the usage of the human-friendly Base32 encoding with an optimized character set, it implements a [BCH code](https://en.wikipedia.org/wiki/BCH_code) that _guarantees detection_ of any error affecting at most 4 characters and has less than a 1 in 10<sup>9</sup> chance of failing to detect more errors.
+
+This RFC proposes a simple and extendable binary serialization for the two address types that is then Bech32 encoded to provide a unique appearance for human-facing applications such as wallets. 
+
+# Detailed design
+
+## Binary serialization
+
+The address format uses a simple serialization scheme which consists of two parts:
+
+   - The first byte describes the type of the address.
+   - The remaining bytes contain the type-specific raw address bytes.
+
+Currently, two kinds of addresses are supported:
+ - W-OTS, which is compatible to the existing legacy IOTA signature scheme and
+ - Ed25519, where the address consists of the BLAKE2b-256 hash of the Ed25519 public key.
+
+They are serialized as follows:
+
+| Type    | First byte | Address bytes                                             |
+| ------- | ---------- | --------------------------------------------------------- |
+| W-OTS   | `0x00`     | 49 bytes: The t5b1 encoded Kerl hash of the W-OTS digest. |
+| Ed25519 | `0x01`     | 32 bytes: The BLAKE2b-256 hash of the Ed25519 public key. |
```

Comment:
Does this mean we are going to use all things defined in Ed25 including keypair generation instead of 243 tris or whatever we use before?

Reply:
I am not sure I fully understand this comment.

In the current proposal RFC https://github.com/iotaledger/protocol-rfcs/pull/18 there are two alternative signature schemes that can be used:
1. Legacy W-OTS 100% compatible to existing private keys, addresses and thus based on 243-trit hashes.
2. Ed25519 exactly as described in [RFC-8032](https://tools.ietf.org/html/rfc8032), which is based entirely on binary private- / public keys.

Reply:
thanks, the second one about if Ed25 is entirely based on binary key pair is what I want to know.